server/base/src/main/java/org/apache/accumulo/server/tabletserver/LargestFirstMemoryManager.java

```diff
@@ -121,18 +122,16 @@ public void remove(Long key) {
     }
   }
 
-  LargestFirstMemoryManager(long maxMemory, int maxConcurrentMincs, int numWaitingMultiplier) {
-    this();
-    this.maxMemory = maxMemory;
-    this.maxConcurrentMincs = maxConcurrentMincs;
-    this.numWaitingMultiplier = numWaitingMultiplier;
+  @Override
+  public void init(ServerConfiguration conf) {
+    throw new IllegalStateException("Deprecated method, no longer should be used.");
```

Comment:
An option for this update would be to call the new init method:
```init(((ServerConfigurationFactory) conf).getServerContext());```

But I put the exception to prevent calling this in Accumulo.  If a user extends this class, I don't think they should be calling this method.  The init currently gets called in TabletServerResourceManager [here](https://github.com/apache/accumulo/blob/94844b3ef88bc9e02b5cb5c8eeff81411ac8221d/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java#L432).

Reply:
Rather than override it or call the other method (which won't do anything, since we never call the deprecated one in the tserver), you could make it a deprecated default method in the interface that throws the exception. The exception message, as well as the deprecation message can say that it is scheduled for removal in 3.0.0. Since default methods don't need to be overridden, people can safely delete their current implementation to prepare for its eventual removal.

However, I'm not sure the replacement accepting a ServerContext is any safer. It's still a leak of an internal type (ServerContext) that is subject to change, into what is effectively SPI. We should probably have some other specific MemoryManagerEnvironment type for this.

Reply:
I like throwing an exception on the default implementation, not sure why I didn't just do that.  

Sounds like this interface might be a good candidate to move to the SPI since it deals more with performance and not user data.

Reply:
I made the change in 1824acd but I made the new method not have a default impl.  My thinking was that since we won't have a way to make them backward compatible, it would be better to error during compilation than runtime in the tserver.

Reply:
Also, I could make the MemoryManager change in a separate PR, since its the only contentious change here.